lib/concurrent_guest_installations.pm

```diff
@@ -112,6 +112,8 @@ sub install_guest_instances {
         else {
             $guest_instances{$_}->guest_installation_run(@_);
         }
+        # abort current guest installation if dry run failed
+        return if $self->{guest_installation_result} eq 'FAILED';
```

Comment:
Are we providing a clearer indication that the installation is being aborted due to a failure?

Reply:
To serve your purpose, which should be checked, `$guest_instances{$_}->{guest_installation_result}` or `$self->{guest_installation_result}`?

What's more important , when multiple vms are being installed, this change will stop further handling for the problematic vm and also LATTER vms. Is it intended too? If I remember correctly, one of the key behavior of concurrent guest installation is to let other vms' installation continue while some fail. That's what are preferred.

At last, suggest to do more careful&comprehensive verifications on this, especially different vm states for multiple vm installation case.

Reply:
> Are we providing a clearer indication that the installation is being aborted due to a failure? 

For guest installation, we don't abort the test if some guests fail. we'd like to continue with other guests and consequent feature tests.

> After this will it throw an exception and terminate the execution or stop the test here?

No, I planed to skip the current guest installation and continue with other guests, but obviously the code is not correct at here. I will correct it.

Reply:
> To serve your purpose, which should be checked, `$guest_instances{$_}->{guest_installation_result}` or `$self->{guest_installation_result}`?

I think you are correct. I need look into it closely.

> What's more important , when multiple vms are being installed, this change will stop further handling for the problematic vm and also LATTER vms. Is it intended too? If I remember correctly, one of the key behavior of concurrent guest installation is to let other vms' installation continue while some fail. That's what are preferred.

I planed to abort the installation for current guest only, but the codes are not correct at here. I am thinking of it should be 'next' instead of 'return'...

> At last, suggest to do more careful&comprehensive verifications on this, especially different vm states for multiple vm installation case.

I will put this change into next PR. Because tests can be run with one PR only, I usually put all changes in it. Especially this kind of code fix to failures which don't always reproduce. They can be merged until they got enough verification runs.

I will mark this commit as WIP and drop it when the PR is going be merged right now. And since the PR for xen test  has been merged, it is capable to verify it on xen tests.